### DISCUSSION TITLE
Error in Regional K Means

### DISCUSSION THREAD

**Discussion by ldeoto** [pysal/spopt#369] [Q&A] [answered]
Hi all, I am new to spopt. I am trying to test the regional-k-means on a polygon .shp and at the moment of solving the model I keep having the following error:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In [41], line 2
      1 model = RegionKMeansHeuristic(ibli.values, 3, rsbr_w)
----> 2 model.solve()

File /usr/local/lib/python3.8/dist-packages/spopt/region/region_k_means.py:171, in RegionKMeansHeuristic.solve(self)
    169 def solve(self):
    170     """Solve the region k-means heuristic."""
--> 171     centroid, label, iters = region_k_means(self.data, self.n_clusters, self.w)
    172     self.labels_ = label
    173     self.centroids_ = centroid

File /usr/local/lib/python3.8/dist-packages/spopt/region/region_k_means.py:79, in region_k_means(X, n_clusters, w)
     77 if neighbors:
     78     d_min = numpy.inf
---> 79     centroid = data[region].mean(axis=0)
     80     for neighbor in neighbors:
     81         d = ((data[neighbor] - centroid) ** 2).sum()

File /usr/local/lib/python3.8/dist-packages/numpy/core/_methods.py:182, in _mean(a, axis, dtype, out, keepdims, where)
    180 ret = umr_sum(arr, axis, dtype, out, keepdims, where=where)
    181 if isinstance(ret, mu.ndarray):
--> 182     ret = um.true_divide(
    183             ret, rcount, out=ret, casting='unsafe', subok=False)
    184     if is_float16_result and out is None:
    185         ret = arr.dtype.type(ret)

TypeError: unsupported operand type(s) for /: 'Polygon' and 'int'
```

Any tips? Thanks!

**Comment by jGaboardi** [comment]
In order to help troubleshoot this we will need both a [MWE](https://en.wikipedia.org/wiki/Minimal_reproducible_example#:~:text=In%20computing%2C%20a%20minimal%20reproducible,to%20be%20demonstrated%20and%20reproduced.) of code that can be copy-pasted and the data you are using as input. 

Judging from your traceback, it appears you are passing in a column of `Polygon` objects. If you take a look at the [Regional K Means tutorial](https://pysal.org/spopt/notebooks/reg-k-means.html#2.-Large-synthetic-example), you'll see that we are generating the weights from geometries, but not actually passing any geometries into `RegionKMeansHeuristic`.

cc @sjsrey

**Reply by ldeoto** [reply, marked as answer]
@jGaboardi thanks for this tip! It worked indeed!
